src/client/Microsoft.Identity.Client/Platforms/Mac/AuthenticationAgentNSWindowController.cs

```diff
@@ -211,7 +214,7 @@ void DecidePolicyForNavigation(WebView webView, NSDictionary actionInformation,
                 }
 
                 Dictionary<string, string> keyPair = CoreHelpers.ParseKeyValueList(query, '&', true, false, null);
-                string responseHeader = DeviceAuthHelper.CreateDeviceAuthChallengeResponseAsync(keyPair).Result;
+                string responseHeader = _deviceAuthManager.CreateDeviceAuthChallengeResponseAsync(keyPair).Result;
```

Comment:
How come you're adding DeviceAuth to a single embedded UI in the whole stack - the one that is probably not used by anybody- MAC? 

1. Can you clarify that the device auth logic needs to occur on the /authroize endpoint? 
2. Was this tested ?

Reply:
I am not sure what is going on here exactly. Mark took some components from ADAl's pkey auth classes and added them here for this class. i am assuming mark tested this when he added it.

Reply:
Pretty sure you have to get rid of this since the devicemanager on MAC does nothing.

Reply:
`CreateDeviceAuthChallengeResponseAsync` return null here, but you do not check for this.